### PR TITLE
Deployments used on 30 Jan Sprint Review

### PR DESCRIPTION
**Pull request by marrobi**
Merge of #1086 #1072 #1048 #1050 #1098 

If prefer to merge rather than individual PRs. 

Is outstanding issues in T-Shirt Sizing and Log Analytics, but can be fixed later if desired.

Outstanding:
- [ ] @marrobi comments in verify logs #1072
- [ ] @marrobi add SSL #1048 
- [x] @marrobi changes timeout to > 60 seconds

### CODE REVIEW

**Comment by marrobi** [comment]
@lawrencegripper not sure of intention of top two checkboxes - we have different issues to track SSL and outstanding log items, 

Testing with oracle jdk images then will be good to merge.

**Comment by marrobi** [comment]
I'm good with this now. @lawrencegripper @jjcollinge want to give it a look over before merge given lots of moving parts.

**Comment by lawrencegripper** [comment]
You've got a fair few armval issues but given the move here #1084 do we care at the moment?

File: opencga-app/app/scripts/azure/arm/appgw/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+        "name": "[variables('applicationGatewayName')]",
+        "apiVersion": "2018-08-01",
+        "location": "[resourceGroup().location]",
+        "properties": {
+            "sku": {
+                "name": "[parameters('skuName')]",
+                "tier":  "[parameters('skuTier')]",
+                "capacity": "[parameters('skuCapacity')]"
+            },
+            "enableHttp2":true,
```
Comment (lawrencegripper) [inline opencga-app/app/scripts/azure/arm/appgw/azuredeploy.json]: We probably need to call out that this won't allow us to do GRPC as it's not HTTP2 all the way through

**Comment by marrobi** [comment]
Wondered what you would say about those! Soooo many schema issues. I'd be tempted to leave them for now.

File: opencga-app/app/scripts/azure/arm/azurebatch/azuredeploy.json [hunk 2]
```diff
@@ -147,7 +188,7 @@
                 "scaleSettings": {
                     "autoScale": {
                         "evaluationInterval": "PT5M",
-                        "formula": "[concat('startingNumberOfVMs = 0;maxNumberofVMs = ', parameters('maxNodeCount'), ';pendingTaskSamplePercent = $PendingTasks.GetSamplePercent(160 * TimeInterval_Second);pendingTaskSamples = pendingTaskSamplePercent < 70 ? startingNumberOfVMs : avg($PendingTasks.GetSample(160 * TimeInterval_Second));$TargetDedicatedNodes=min(maxNumberofVMs, pendingTaskSamples);')]"
+                        "formula": "[concat('startingNumberOfVMs = 1;maxNumberofVMs = ', parameters('maxNodeCount'), ';pendingTaskSamplePercent = $PendingTasks.GetSamplePercent(160 * TimeInterval_Second);pendingTaskSamples = pendingTaskSamplePercent < 70 ? startingNumberOfVMs : avg($PendingTasks.GetSample(160 * TimeInterval_Second));$TargetDedicatedNodes=max(1,min(maxNumberofVMs, pendingTaskSamples));')]"
```
Comment (lawrencegripper) [inline opencga-app/app/scripts/azure/arm/azurebatch/azuredeploy.json]: Did we want this set as a param and changed by the TShirt sizes?

File: opencga-app/app/scripts/azure/arm/daemonvm/azuredeploy.json [hunk 3]
```diff
@@ -120,14 +138,14 @@
...
         "daemonCmd": "[concat('docker run -d --restart=always --mount type=bind,src=/media/primarynfs/conf,dst=/opt/opencga/conf,readonly --mount type=bind,src=/media/primarynfs/sessions,dst=/opt/opencga/sessions --mount type=bind,src=/media/primarynfs/variants,dst=/opt/opencga/variants -e OPENCGA_PASS=',parameters('openCgaAdminPassword'),' ',parameters('daemonContainerImage'))]",
+        "logAnalyticsAgent": "[concat('docker run --privileged -d -v /var/run/docker.sock:/var/run/docker.sock -v /var/log:/var/log -v /var/lib/docker/containers:/var/lib/docker/containers -e WSID=',parameters('logAnalyticsWorkSpaceID'),' -e KEY=',parameters('logAnalyticsWorkSpaceKey'),' -p 127.0.0.1:25225:25225 -p 127.0.0.1:25224:25224/udp --name=omsagent -h=`hostname` --restart=always microsoft/oms')]",
```
Comment (lawrencegripper) [inline opencga-app/app/scripts/azure/arm/daemonvm/azuredeploy.json]: What does the logging agent use the ports for? Seems odd

Comment (marrobi) [reply opencga-app/app/scripts/azure/arm/daemonvm/azuredeploy.json]: believe its to retrieve syslogs from the host.

File: opencga-app/app/scripts/azure/arm/appgw/azuredeploy.json (hunk 1 above)
Comment (marrobi) [reply opencga-app/app/scripts/azure/arm/appgw/azuredeploy.json]: shall I add a comment to the json? :-)

File: opencga-app/app/scripts/azure/arm/azurebatch/azuredeploy.json (hunk 2 above)
Comment (marrobi) [reply opencga-app/app/scripts/azure/arm/azurebatch/azuredeploy.json]: We did, I'll add that now I guess, unless best to get this in and then fix that. Might be easier as need to get this merged.

File: opencga-app/app/scripts/azure/arm/daemonvm/azuredeploy.json (hunk 3 above)
Comment (lawrencegripper) [reply opencga-app/app/scripts/azure/arm/daemonvm/azuredeploy.json]: Looks to me like its something to do with this [cisco logging thing](https://github.com/Microsoft/OMS-Agent-for-Linux/blob/master/docs/Security-Events-Preview-Configuration.md) as I'd have thought syslog is read from the `var/log` folder mount.

File: opencga-app/app/scripts/azure/arm/appgw/azuredeploy.json (hunk 1 above)
Comment (lawrencegripper) [reply opencga-app/app/scripts/azure/arm/appgw/azuredeploy.json]: Maybe just something to track as an issue so everyone is aware when we come to GRPC stuff?

File: opencga-app/app/scripts/azure/arm/azurebatch/azuredeploy.json (hunk 2 above)
Comment (lawrencegripper) [reply opencga-app/app/scripts/azure/arm/azurebatch/azuredeploy.json]: Happy either way.

File: opencga-app/app/scripts/azure/arm/daemonvm/azuredeploy.json (hunk 3 above)
Comment (marrobi) [reply opencga-app/app/scripts/azure/arm/daemonvm/azuredeploy.json]: "By default, the OMS Agent for Linux receives events from the syslog daemon over UDP. In cases where a Linux machine is expected to collect a high volume of syslog events, such as when a Linux agent is receiving events from other devices, the configuration should be modified to use TCP transport between the syslog daemon and OMS agent."  https://github.com/Microsoft/OMS-Agent-for-Linux/blob/master/docs/Security-Events-Preview-Configuration.md

File: opencga-app/app/scripts/azure/arm/azurebatch/azuredeploy.json (hunk 2 above)
Comment (marrobi) [reply opencga-app/app/scripts/azure/arm/azurebatch/azuredeploy.json]: #1104

File: opencga-app/app/scripts/azure/arm/appgw/azuredeploy.json (hunk 1 above)
Comment (marrobi) [reply opencga-app/app/scripts/azure/arm/appgw/azuredeploy.json]: added comment on #1027

File: opencga-app/app/scripts/azure/arm/daemonvm/azuredeploy.json (hunk 3 above)
Comment (lawrencegripper) [reply opencga-app/app/scripts/azure/arm/daemonvm/azuredeploy.json]: But we haven't done #3 on this list https://github.com/Microsoft/OMS-Agent-for-Linux/blob/master/docs/Security-Events-Preview-Configuration.md#detailed-configuration so it won't be used in our case or did I miss us configuring these bits?

Comment (lawrencegripper) [reply opencga-app/app/scripts/azure/arm/daemonvm/azuredeploy.json]: Looks like these are required but no idea how these get handled under the covers.

**Comment by marrobi** [comment]
@lawrencegripper armval looks ok, are you happy with the exclusions?

**Comment by lawrencegripper** [comment]
@marrobi I've taken a look at the `armval` rules and they run the risk of masking valid errors. Let me have a look at the file with the new ignore syntax and I'll propose some changes to the ignore - if that's ok with you?

**Comment by marrobi** [comment]
Go for it, as you know I was getting frustrated!

**Comment by lawrencegripper** [comment]
Deployment succeeded with latest merge of code and ARMVAL work, not sure I have the latest images but happy to merge and work from this now.

**Pull request merged**